src/builtin.c

```diff
@@ -359,7 +359,8 @@ static jv f_multiply(jq_state *jq, jv input, jv a, jv b) {
       num = a;
     }
     jv res = jv_null();
-    int n = jv_number_value(num);
+    double d = jv_number_value(num);
+    int n = 0.0 < d && d < 1.0 ? 1 : d;
```

Comment:
Questions while we're here:

 - if `("string" * 0) == null`, why should `("string" * 0.1) == "string"`?
 - shouldn't `("string") * 0 == ""`?  `null` seems very wrong
 - currently multiplying by a negative number, `nan`, `-nan`, `infinite`, or `-infinite` yields `null` -- why not an error, or `empty`?  (well, `empty` would be harder to arrange here, so let's not go there)